lightning/src/onion_message/messenger.rs

```diff
@@ -1156,21 +1146,38 @@ where
 		self.offers_handler = offers_handler;
 	}
 
-	/// Sends an [`OnionMessage`] with the given `contents` to `destination`.
-	///
-	/// See [`OnionMessenger`] for example usage.
+	/// Sends an [`OnionMessage`] based on its [`MessageSendInstructions`].
 	pub fn send_onion_message<T: OnionMessageContents>(
-		&self, contents: T, destination: Destination, reply_path: Option<BlindedMessagePath>
+		&self, contents: T, instructions: MessageSendInstructions,
 	) -> Result<SendSuccess, SendError> {
-		self.find_path_and_enqueue_onion_message(
-			contents, destination, reply_path, format_args!("")
-		)
+		self.send_onion_message_internal(contents, instructions, format_args!(""))
 	}
 
-	fn find_path_and_enqueue_onion_message<T: OnionMessageContents>(
-		&self, contents: T, destination: Destination, reply_path: Option<BlindedMessagePath>,
-		log_suffix: fmt::Arguments
+	fn send_onion_message_internal<T: OnionMessageContents>(
+		&self, contents: T, instructions: MessageSendInstructions, log_suffix: fmt::Arguments,
 	) -> Result<SendSuccess, SendError> {
+		let (destination, reply_path) = match instructions {
+			MessageSendInstructions::WithSpecifiedReplyPath { destination, reply_path } =>
+				(destination, Some(reply_path)),
+			MessageSendInstructions::WithReplyPath { destination, context }|
+			MessageSendInstructions::ForReply { instructions: ResponseInstruction { destination, context: Some(context) } } => {
+				match self.create_blinded_path(context) {
+					Ok(reply_path) => (destination, Some(reply_path)),
+					Err(err) => {
+						log_trace!(
+							self.logger,
+							"Failed to create reply path {}: {:?}",
+							log_suffix, err
+						);
+						return Err(err);
+					}
+				}
+			},
+			MessageSendInstructions::WithoutReplyPath { destination }|
```

Comment:
Could you put the `|` on the next line and space accordingly. It's a bit difficult to read like this and rustfmt will eventually change it.